### PR TITLE
Extract actions on new pull request from models to pulls service and move code.gitea.io/gitea/modules/pull to code.gitea.io/gitea/services/pull

### PR DESCRIPTION
**Pull request by lunny**
As title.

### CODE REVIEW

File: services/pulls/pulls.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/log"
+	api "code.gitea.io/gitea/modules/structs"
+)
+
+// NewPullRequest creates new pull request with labels for repository.
+func NewPullRequest(repo *models.Repository, pull *models.Issue, labelIDs []int64, uuids []string, pr *models.PullRequest, patch []byte, assigneeIDs []int64) (err error) {
```
Comment (lafriks) [inline services/pulls/pulls.go]:
```suggestion
func NewPullRequest(repo *models.Repository, pull *models.Issue, labelIDs []int64, uuids []string, pr *models.PullRequest, patch []byte, assigneeIDs []int64) error {
```

File: services/pulls/pulls.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	if err = models.NewPullRequest(repo, pull, labelIDs, uuids, pr, patch, assigneeIDs); err != nil {
```
Comment (lafriks) [inline services/pulls/pulls.go]:
```suggestion
	if err := models.NewPullRequest(repo, pull, labelIDs, uuids, pr, patch, assigneeIDs); err != nil {
```

File: services/pulls/pulls.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		return
```
Comment (lafriks) [inline services/pulls/pulls.go]:
```suggestion
		return err
```

File: services/pulls/pulls.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	}
+
+	if err = models.NotifyWatchers(&models.Action{
```
Comment (lafriks) [inline services/pulls/pulls.go]:
```suggestion
	if err := models.NotifyWatchers(&models.Action{
```

File: services/pulls/pulls.go [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+		Repo:      repo,
+		IsPrivate: repo.IsPrivate,
+	}); err != nil {
+		log.Error("NotifyWatchers: %v", err)
+	}
+
+	pr.Issue = pull
+	pull.PullRequest = pr
+	mode, _ := models.AccessLevel(pull.Poster, repo)
+	if err = models.PrepareWebhooks(repo, models.HookEventPullRequest, &api.PullRequestPayload{
```
Comment (lafriks) [inline services/pulls/pulls.go]:
```suggestion
	if err := models.PrepareWebhooks(repo, models.HookEventPullRequest, &api.PullRequestPayload{
```

**Review by lafriks — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
@lafriks done.

File: routers/api/v1/repo/pull.go [hunk 6]
```diff
@@ -20,6 +20,7 @@ import (
 	api "code.gitea.io/gitea/modules/structs"
 	"code.gitea.io/gitea/modules/timeutil"
 	milestone_service "code.gitea.io/gitea/services/milestone"
+	pull_service "code.gitea.io/gitea/services/pulls"
```
Comment (lafriks) [inline routers/api/v1/repo/pull.go]: import declared with same name as import above:
```
pull_service "code.gitea.io/gitea/modules/pull"
```

File: services/pulls/pulls.go [hunk 7]
```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package pulls
```
Comment (lafriks) [inline services/pulls/pulls.go]: It should probably be named `pull` to have same naming as it already is for `modules/pull`

**Review by lafriks — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
@lafriks done and moved `code.gitea.io/gitea/modules/pull` to `code.gitea.io/gitea/services/pull`

**Review by guillep2k — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Comment by lunny** [comment]
make L-G-T-M work

**Pull request merged**